### ISSUE TITLE
React wrapper

### ISSUE THREAD

**Issue by jussi-kalliokoski** [chjj/blessed#112]
Just throwing an idea in the air... It would be really cool to be able to define terminal UIs in a React-style way, and this library seems like a perfect fit for wrapping into a React-style API (in a separate module of course).

**Comment by chjj** [comment]
This is definitely one of the directions I want to see blessed go. I have a few in mind actually, just off the top of my head:

1. Create a markup language for blessed (already done, and pretty cool: https://github.com/kevinhikaruevans/blessedoo)
2. Use blessed as a rendering backend for jsdom, allowing us to create an insanely good terminal web browser (have you ever seen a terminal browser with hover effects and hoverable text boxes?).
3. Wrap blessed with any other APIs that have a concept of rendering, such as this.

**Comment by chjj** [comment]
Of course, I wouldn't include this in core blessed. These are things that should be separate in my opinion.

**Comment by secrettriangle** [comment]
:+1:

**Comment by taneltm** [comment]
I've been having the same idea since I first learned about blessed - More specifically, blessing backbone/marionette.
Blessedoo should make it much easier!

**Comment by Yomguithereal** [comment]
Hello everyone. With the latest changes in `react@0.14.0` and the separation of the rendering logic from the components' one, I started a little proof of concept of `react-blessed` [here](https://github.com/Yomguithereal/react-blessed). It does not much for the time being but if you like the idea, we can start working on that an extend the features of the renderer quite easily.

**Comment by chjj** [comment]
@Yomguithereal, good job. I wasn't aware React was modularized in that way. That makes things a lot easier. That's one idea. I think it's worth pursuing.

But the ultimate goal I have in my head should be to combine JSDOM with a CSSOM module (anyone know of a good headless CSSOM?) and use Blessed for the rendering backend. Then we could use whatever frontend framework we wanted without changing a thing.

**Comment by Yomguithereal** [comment]
What about [cssom](https://www.npmjs.com/package/cssom) the npm package (sorry if this is not headless and for some stupid reasons I missed it).

Beotian question: is jsdom rendering backend modifiable without a kind of fork?

**Comment by Yomguithereal** [comment]
@chjj, I just started experimenting with the latest jsdom version (which packs a partiall CSSOM) and am trying to use facebook's [css-layout](https://github.com/facebook/css-layout) module on the resultant DOM to see whether the resultant layout is of sufficient quality

**Comment by cancerberoSgx** [comment]
Just JSX no react, one file implementation with types and supporting, FunctionElement, ClassElement, function attributes and things like {arr.map(a=><text....>} or {condition && <box...} 

 * implementation: https://github.com/cancerberoSgx/typescript-ast-explorer/blob/dba7859e301d9b86c065a2881151471f63c89417/src/blessed/jsx.ts
 * test: https://github.com/cancerberoSgx/typescript-ast-explorer/blob/dba7859e301d9b86c065a2881151471f63c89417/spec/blessed/jsxTest2.tsx
